### PR TITLE
Switch to cibuildwheel and build macOS wheels

### LINKED ISSUE #42: Install pyonmttok on macOS?
Hi,

I'm trying to install on macOS but it fails with the message

`Could not find a version that satisfies the requirement pyonmttok (from versions: )`

I have macOS's python2.7 and homebrew python3.7. Searching with pip finds the package though.

### PR DESCRIPTION
**Pull request by keichi**
Closes #42

This PR is largely based on @filips123's efforts at #136, integrated with the current GitHub Actions CI script.

Note that I had to move the extension to a subpackage (`pyonmttok._ext`) due to a limitation in the delocate utility (see https://github.com/matthew-brett/delocate#external-library-is-not-bundled-into-wheel and https://github.com/matthew-brett/delocate/issues/72).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,21 +63,19 @@ jobs:
 
       - name: Build wheels
         run: |
-          docker run -v $PWD:/root -w /root quay.io/pypa/manylinux1_x86_64 bash /root/bindings/python/tools/build_wheel.sh
```
Comment (guillaumekln) [inline .github/workflows/ci.yml:47]: Can you remove the files `tools/build_wheel.sh` and `tools/build_wheel_macosx.sh` as part of this change?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -44,21 +63,19 @@ jobs:
...
-        run: |
-          python -m pytest bindings/python/test/test.py
+          python -m pip install cibuildwheel==1.7.2
+          python -m cibuildwheel bindings/python --output-dir wheelhouse
+        env:
+          CIBW_BEFORE_ALL: bindings/python/tools/prepare_build_environment.sh
+          CIBW_BEFORE_BUILD: pip install pybind11==2.6.0
+          CIBW_TEST_COMMAND: pytest {project}/bindings/python/test/test.py
+          CIBW_TEST_REQUIRES: pytest
+          CIBW_SKIP: cp27-* pp* *i686
```
Comment (guillaumekln) [inline .github/workflows/ci.yml:73]: Previous versions of the Tokenizer used `manylinux1` but it looks like `cibuildwheel` uses `manylinux2010` by default. Can we try building under `manylinux1` to not require users to upgrade their pip version? It seems the following should be enough:

```yaml
CIBW_MANYLINUX_X86_64_IMAGE: manylinux1
```

**Review by guillaumekln — COMMENTED** [review]
Thank you for working on this! The changes are looking good.

**Comment by keichi** [comment]
Thanks for the review, I've addressed your comments.

**Comment by guillaumekln** [comment]
Thanks!

**Pull request merged**